### PR TITLE
schedule.mutation() should always run tasks async (closes #33)

### LINKED ISSUE #33: schedule.mutation() should always run tasks async
We're inconsistent right now in that if the mutation is blocked the task will be performed asynchronously, whereas if it's not, it will performed synchronously.

This risks introducing subtle, hard to reproduce, bugs into app codebases.

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@etiennesegonzac r?

**Comment by wilsonpage** [comment]
Out of interest, what do you mean by 'naive' in the context of this library?

**Comment by etiennesegonzac** [comment]
> Out of interest, what do you mean by 'naive' in the context of this library?

The goal is to approximate how terribly things will perform without the scheduler.
It was super useful in the beginning to do before / after demos.

File: lib/dom-scheduler.js [hunk 1]
```diff
@@ -294,7 +294,7 @@
             resolve: resolve
           });
         } else {
-          block();
+          Promise.resolve().then(block);
           resolve();
```
Comment (etiennesegonzac) [inline lib/dom-scheduler.js]: I know technically it should dequeued after but it's a bit frightening to read at first.
`Promise.resolve().then(block).then(resolve)` reads a bit better IMO

File: test/dom-scheduler-test.js [hunk 2]
```diff
@@ -272,6 +272,15 @@ suite('DomScheduler >', function() {
           done();
         });
     });
+
+    test.only('it always executes the task async', function() {
```
Comment (etiennesegonzac) [inline test/dom-scheduler-test.js]: let's not forget the `only` here :)

**Pull request merged**